chisel/src/logger.rs

```diff
@@ -0,0 +1,30 @@
+static mut LOG_LEVEL: i32 = 0;
+
+#[macro_export]
+macro_rules! chisel_debug {
+    ($lvl:expr, $($arg:tt)*) => {
+        crate::logger::Logger::with_global_level().log($lvl, &format!($($arg)*));
+    }
+}
+
+/// Simple logging utility struct.
+pub struct Logger(i32);
+
+impl Logger {
+    pub fn with_global_level() -> Self {
+        unsafe { Logger(LOG_LEVEL) }
```

Comment:
Does this needs any synchronisation or something since it accesses a global?

Reply:
Or at the bare minimum a big fat warning stating that this could use a mutex, but the CLI is structured in a way it should not.

Reply:
When we move things into the library I will, but for now we can safely assume that the CLI is all single thread. Will label it.